lib/models/device.coffee

```diff
@@ -2170,6 +2186,140 @@ getDeviceModel = (deps, opts) ->
 				body: should_be_running__release: null
 		.asCallback(callback)
 
+	###*
+	# @summary Check whether the provided device can update to the target os version
+	# @name _checkOsUpdateTarget
+	# @private
+	# @function
+	# @memberof balena.models.device
+	#
+	# @description
+	# utility method exported for testability
+	#
+	# @param {Object} device - A device object
+	# @param {String} targetOsVersion - semver-compatible version for the target device
+	# @throws Exception if update isn't supported
+	# @returns {undefined}
+	###
+	exports._checkOsUpdateTarget = ({ uuid, device_type, is_online, os_version, os_variant }, targetOsVersion) ->
+		if not is_online
+			throw new Error("The device is offline: #{uuid}")
+
+		if not uuid
+			throw new Error('The uuid of the device is not available')
```

Comment:
Is this a message shown to the user? Just wondering about the wording, and in what case this would happen. Is it like "Device not found with this UUID?" Also, should it come before "not is_online", a the UUID check is a more basic test, right?

Reply:
> Is this a message shown to the user?

No. Since `_checkOsUpdateTarget()` isn't a public/documented method, that is called by `startOsUpdate()`, this check is there so that we will notice if the device object passed in is missing the `uuid`, which is required for the rest of the errors. In short, this will only show up in the tests cases, in case we do a bad refactor.

> should it come before "not is_online", a the UUID check is a more basic test, right?

Yes, that's correct :+1: